### PR TITLE
VZ-V4821 upgrade test to check if workloads are running

### PR DESCRIPTION
**Pull request by pfmackin**
This is a post-upgrade test to make sure at least 1 replica is running for all of our platform workloads

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by michael-cico — APPROVED** [review]

File: tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go [hunk 1]
```diff
@@ -138,3 +174,120 @@ var _ = t.Describe("istioctl verify-install", func() {
...
+			t.Entry("Checking Deployment oam-kubernetes-runtime", constants.VerrazzanoSystemNamespace, oam.ComponentName, "oam-kubernetes-runtime"),
+			t.Entry("Checking Deployment verrazzano-console", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "verrazzano-console"),
+			t.Entry("Checking Deployment vmi-system-grafana", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-grafana"),
+			t.Entry("Checking Deployment verrazzano-console", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "verrazzano-console"),
+			t.Entry("Checking Deployment vmi-system-kibana", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-kibana"),
+			t.Entry("Checking Deployment vmi-system-prometheus-0", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-prometheus-0"),
+			t.Entry("Checking Deployment weblogic-operator", constants.VerrazzanoSystemNamespace, weblogic.ComponentName, "weblogic-operator"),
+
+			t.Entry("Checking Deployment cert-manager", certmanager.ComponentNamespace, certmanager.ComponentName, "cert-manager"),
```
Comment (bobdonat) [inline tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:210]: won't this sometimes falsely match cert-manager-cainjector or cert-manager-webhook

File: tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go [hunk 2]
```diff
@@ -138,3 +174,120 @@ var _ = t.Describe("istioctl verify-install", func() {
...
+			t.Entry("Checking Deployment istio-egressgateway", compistio.IstioNamespace, compistio.ComponentName, "istio-egressgateway"),
+
+			t.Entry("Checking Deployment vmi-system-kiali", constants.VerrazzanoSystemNamespace, kiali.ComponentName, "vmi-system-kiali"),
+
+			t.Entry("Checking Deployment mysql", mysql.ComponentNamespace, mysql.ComponentName, "mysql"),
+
+			t.Entry("Checking Deployment ingress-controller-ingress-nginx-controller", nginx.ComponentNamespace, nginx.ComponentName, "ingress-controller-ingress-nginx-controller"),
+			t.Entry("Checking Deployment ingress-controller-ingress-nginx-defaultbackend", nginx.ComponentNamespace, nginx.ComponentName, "ingress-controller-ingress-nginx-defaultbackend"),
+
+			t.Entry("Checking Deployment rancher", rancher.ComponentNamespace, rancher.ComponentName, "rancher"),
```
Comment (bobdonat) [inline tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:227]: same comment as above.  won't this falsely match rancher-webhook sometimes

File: tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go [hunk 3]
```diff
@@ -138,3 +174,120 @@ var _ = t.Describe("istioctl verify-install", func() {
...
+			t.Entry("Checking Deployment verrazzano-authproxy", constants.VerrazzanoSystemNamespace, authproxy.ComponentName, "verrazzano-authproxy"),
+			t.Entry("Checking Deployment coherence-operator", constants.VerrazzanoSystemNamespace, coherence.ComponentName, "coherence-operator"),
+			t.Entry("Checking Deployment oam-kubernetes-runtime", constants.VerrazzanoSystemNamespace, oam.ComponentName, "oam-kubernetes-runtime"),
+			t.Entry("Checking Deployment verrazzano-console", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "verrazzano-console"),
+			t.Entry("Checking Deployment vmi-system-grafana", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-grafana"),
+			t.Entry("Checking Deployment verrazzano-console", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "verrazzano-console"),
+			t.Entry("Checking Deployment vmi-system-kibana", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-kibana"),
+			t.Entry("Checking Deployment vmi-system-prometheus-0", constants.VerrazzanoSystemNamespace, verrazzano.ComponentName, "vmi-system-prometheus-0"),
```
Comment (bobdonat) [inline tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:207]: does this need to check for es-ingest and es-data too

Comment (bobdonat) [reply tests/e2e/upgrade/post-upgrade/verify/verify_restart_test.go:207]: and VMO

**Pull request merged**